Aetherium/Artifacts/ArtifactOfRegression.cs

```diff
@@ -141,36 +148,44 @@ public override void Hooks()
 
         private void RegressAIToLowerForm(On.RoR2.CharacterAI.BaseAI.orig_OnBodyDeath orig, RoR2.CharacterAI.BaseAI self, CharacterBody characterBody)
         {
-            if (NetworkServer.active && ArtifactEnabled)
+            CharacterMaster master = self.master;
+            if (ArtifactEnabled && NetworkServer.active && IsExisting(master) && IsAnEnemy(master))
             {
-                if (self.master && self.master.IsDeadAndOutOfLivesServer() && (self.master.teamIndex == TeamIndex.Monster || self.master.teamIndex == TeamIndex.Lunar))
+                var masterName = master.name.Replace("(Clone)", "");
+                if (RegressionLookup.ContainsKey(masterName))
                 {
-                    var masterName = self.name.Replace("(Clone)", "");
-                    if (RegressionLookup.ContainsKey(masterName))
+                    foreach (var children in RegressionLookup[masterName].children)
                     {
-                        var regressionMasterName = RegressionLookup[masterName].Item1;
-                        var regressionSpawnAmount = RegressionLookup[masterName].Item2;
-
-                        var masterPrefab = Resources.Load<GameObject>($"Prefabs/CharacterMasters/{regressionMasterName}");
-
-                        if (masterPrefab)
+                        GameObject childMasterPrefab;
+                        string childName = children.Key;
+                        RegressData.ChildData childData = children.Value;
+                        if (childData.resource)
                         {
-                            for(int i = 0; i < regressionSpawnAmount; i++)
+                            childMasterPrefab = childData.resource;
+                        }
+                        else
+                        {
+                            childMasterPrefab = Resources.Load<GameObject>($"Prefabs/CharacterMasters/{childName}");
```

Comment:
can we not do this in Store if the parameter is null?

Reply:
I'm not sire I follow you on this one. What do you mean?

Reply:
Down yonder at https://github.com/KomradeSpectre/AetheriumMod/pull/45/files#diff-d813d1c637412df474aa12ef558985e797d3d4660af66b01f7b6e6fa4ebc89d2R272 you have a Store overload with 2 parameters, instead of passing null as the 3rd param you could just do the Resources.Load call there couldn't you?

Reply:
I get it now. Yes I can move the actual resource there instead.

Reply:
Done.